operator/iamkeycontroller/observe.go

```diff
@@ -29,20 +25,20 @@ func (p *IAMKeyPipeline) Observe(ctx context.Context, mg resource.Managed) (mana
 	iamKey := fromManaged(mg)
 	// to manage state of new and old keys I need other variable, this is why this annotation is set
 	// otherwise observation fails for one of key types
-	if _, exists := iamKey.Annotations["newKeyType"]; exists {
```

Comment:
Won't this break legacy keys?

Reply:
I think Bigli removed them all, we only had legacy keys for ourselves, there were no customers using old keys.

Reply:
Yes it will, but I checked and there aren't any legacy keys in use for appcat